### ISSUE TITLE
Recurrent Support

### ISSUE THREAD

**Issue by chinux23** [aigamedev/scikit-neuralnetwork#105]
I love your toolkit, it's wonderful and super easy to use. I am wondering if you plan to add support for recurrent neural network?

**Comment by alexjc** [comment]
Yes, we're thinking about it :-)  Would you want the API to be different (better suited to temporal data) or the same from what scikit-learn supports?

**Comment by chinux23** [comment]
Glad to hear that :+1:  I am not sure what API would be like for temporal data. But it seems intuitive at least for me if it's like scikit-learn.

**Comment by chinux23** [comment]
I found a python2 recurrent project here: http://pyneurgen.sourceforge.net/recurrent.html
I am not sure if this could be helpful in design the API anyway.

**Comment by alexjc** [comment]
A more temporal API would include a `step()` function that can take infinite sequences of data. With `scikit-learn` you limit the API to something finite and fixed length.

**Comment by ssamot** [comment]
@alexjc 
My two cents. Real time recurrent neural networks are hard to get working in python - I am not aware of a theano implementation that actually works. A temporal API would accept some 3D structure  (e.g. `n_timestep, n_samples, n_inputs' and once trained the network would be able to accept a similar structure that accepts the whole temporal sequence - this is how most packages do it. 

Not sure why this is not flagged as a greater problem - it will look much deeper into it in the following months

**Comment by alexjc** [comment]
Should be possible now, here's how it would work:
http://lasagne.readthedocs.org/en/latest/modules/layers/recurrent.html

We'll probably just add a `Recurrent` layer type like `Convolution` and have it take `rnn | gru | lstm` as a type, then connect everything up via Lasagne automatically :-)

**Comment by alexjc** [comment]
#195 lets you add any native Lasagne layer to sknn, so you can use this to inject LSTM layers for example. If you try this, let me know so I can look at the code and see how best to abstract it ;-)

**Comment by pGit1** [comment]
> #195 lets you add any native Lasagne layer to sknn, so you can use this to inject LSTM layers for example. If you try this, let me know so I can look at the code and see how best to abstract it ;-)

Has anyone tried this? Having difficulty with lasagne Recurrent Layer myself.
